retrofit/src/main/java/retrofit2/ServiceMethod.java

```diff
@@ -282,6 +282,8 @@ private void parseHttpMethodAndPath(String httpMethod, String value, boolean has
         throw methodError("Only one HTTP method is allowed. Found: %s and %s.",
             this.httpMethod, httpMethod);
       }
+      if (null == httpMethod) return;
```

Comment:
if we go through the code in the ServiceMethod build after parseMethodAnnotations , we are checking if httpMethod is null or not and then throwing the exception , so if we check at parseHttpMethodPath itself , we don't need to do value parsing.